README.md

```diff
@@ -12,11 +16,322 @@ Install `reconfix` by running:
 $ npm install --save reconfix
 ```
 
+Documentation
+-------------
+
+Reconfix can be hard to grok. We'll discuss a concrete example in depth in
+order to ease the steep learning curve.
+
+The Raspberry Pi reads system configuration parameters from a file called
+[`/boot/config.txt`][rpi-config-txt], which lives in the first partition of the
+image. This file contains a set of key-value pairs following the [ini] file
+format. For example:
+
+```txt
+gpu_mem=192
+hdmi_drive=2
+hdmi_mode=82
+```
+
+Lets pretend we wrote a YouTube video player application for our Raspberry Pi,
+which also reads certain application configuration from a file called
+`/etc/pitube.json`. Here's an example of our ficticious file:
+
+```json
+{
+  "user": "johndoe",
+  "apiKey": "xxxx-yyyy-zzzz",
+  "quality": 720,
+  "fullscreen": true,
+  "annotations": false
+}
+```
+
+Finally, lets say we've configured our Raspberry Pi to read network
+configuration from a [NetworkManager][networkmanager] file called
+`/etc/network` that looks something like this:
+
+```
+[home]
+Type = wifi
+Name = mynetwork
+Passphrase = secret
+Nameservers = 8.8.8.8,8.8.4.4
+```
+
+We want to build a tool that allows us to read configuration from all the files
+we care about, modify it, and write it back, but the problem is that the
+settings live in different files, which follow different formats. We can of
+course write custom code to handle this small example, but Reconfix allow us to
+tackle the problem in a declarative way.
+
+### Phases
+
+In order to provide bi-directional transformation of data, Reconfix makes use
+of different transformation phases.
+
+#### Dry JSON
+
+This phase consists in a high-level configuration state out of all the files we
+care about. Based on the example we outlined at the beginning of this section, we could design our dry JSON to look something like this:
```

Comment:
This line is a few hundred characters long, while others seem to be wrapped earlier. Could just be the GitHub viewer though.